GliaWidgets/CallVisualizer/CallVisualizer.swift

```diff
@@ -0,0 +1,51 @@
+import Foundation
+import UIKit
+import SalemoveSDK
+
+public final class CallVisualizer {
+
+    private let coreSdk: CoreSdkClient
+
+    init(coreSdk: CoreSdkClient) {
+        self.coreSdk = coreSdk
+    }
+
+    private func presentOnTopViewController(viewController: UIViewController) {
+        if let rootViewController = UIApplication.shared.keyWindow?.rootViewController {
+            rootViewController.present(viewController, animated: false)
+        }
+    }
+
+    public func requestVisitorCode(completion: @escaping (Result<VisitorCode, Error>) -> Void) {
+        _ = coreSdk.requestVisitorCode(completion)
+    }
+
+    public func requestVisitorCodeViewController(uiConfig: RemoteConfiguration? = nil) -> UIViewController {
+        let vc = VisitorCodeViewController(viewType: .alert, uiConfig: uiConfig)
+        vc.modalPresentationStyle = .overFullScreen
+        vc.modalTransitionStyle = .crossDissolve
+        return vc
+    }
+
+    public func presentVisitorCodeViewController(uiConfig: RemoteConfiguration? = nil) {
+        presentOnTopViewController(
+            viewController: requestVisitorCodeViewController(uiConfig: uiConfig)
+        )
+    }
+
+    public func layoutVisitorCodeView(
```

Comment:
Do we need this method?
I mean as far as I understand this ads an additional option to the Integrators on how to attach our visitor code view to the screen. And that is generally nice. But the complexity that it solves for the Integrator is minimum (at least that is how I understood it). They say that it is a bad practice for a library to have too many methods
1. Add the complexity of maintenance (docs, implementation in case of changes)
2. Add a possibility of errors. In this case, there is some logic for removing subviews that needs to be mentioned in docs, and if I understood this code it is not very flexible for dynamic resizing of the container. Probably more thing might go wrong
3. There is a bit more to it

Reply:
fixed [7544f3f](https://github.com/salemove/ios-sdk-widgets/pull/448/commits/7544f3fe501c435883d4e574a6084488b33a46f1)